lib/x11utils.pm

```diff
@@ -322,4 +324,34 @@ sub untick_welcome_on_next_startup {
     }
 }
 
+=head2 workaround_boo1170586
+
+ workaround_boo1170586();
+
+Kill broken opensuse-welcome window and restart it properly to workaround boo#1170586.
+
+=cut
+sub workaround_boo1170586 {
+    record_soft_failure('opensuse-welcome auto-start is broken, boo#1170586');
+    x11_start_program('killall /usr/bin/opensuse-welcome', target_match => 'generic-desktop');
+    x11_start_program('opensuse-welcome');
```

Comment:
please add a `target_match` parameter here so that `x11_start_program` can more quickly decide when it can finish

Reply:
This is implicit, as the default behavior.

Reply:
You are right, I forgot my own code :)